### PR TITLE
doc: add info on building PoCL on Android

### PR DESCRIPTION
**Pull request by RABijl**
additions:
* android topic
* update proxy docs with new features
* remove old remote build instructions for Android and point towards android page.
* also includes reference to original libopencl-stub project

### CODE REVIEW

File: doc/sphinx/source/android.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+.. _android-label:
+
+Using PoCL on Android
+=====================
+
+It is possible to build and use PoCL on Android. However, The the setup requires a number of options to be set.
```
Comment (pjaaskel) [inline doc/sphinx/source/android.rst:6]:
```suggestion
It is possible to build and use PoCL on Android. However, the setup requires a number of options to be set.
```

File: doc/sphinx/source/android.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+To see an example project, have a look at the `PoCL-R Reference Android Java Client <https://github.com/cpc/PoCL-R-Reference-Android-Java-Client>`_ .
+This Reference app uses both the Proxy and Remote device in its example apps. It also builds a custom version of JOCL so
```
Comment (pjaaskel) [inline doc/sphinx/source/android.rst:8]: Can you add internal links to proxy and remote docs?

Comment (pjaaskel) [reply doc/sphinx/source/android.rst:8]: Also link to JOCL github/web?

File: doc/sphinx/source/android.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+that PoCL can be used in Java instead of calling C code using the Java Native Interface (jni). These guidelines assume
+that Android studio is used as an IDE, but it should be possible do something similar with a different IDE. It is also
```
Comment (pjaaskel) [inline doc/sphinx/source/android.rst:10]:
```suggestion
that Android studio is used as an IDE, but it should be possible to do something similar with a different IDE. It is also
```

File: doc/sphinx/source/android.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+For newer Android versions (Android 12 and up), you also need to add::
+
+        <uses-native-library
+            android:name="libOpenCL.so"
+            android:required="false" />
+
+to the ``<applications>`` element of the `AndroidManifest.xml`
+
+Once you know that you the phone comes with an OpenCL library, it possible to use the proxy device. To build the proxy device add the
```
Comment (pjaaskel) [inline doc/sphinx/source/android.rst:59]:
```suggestion
Once you know that your phone comes with an OpenCL library, it possible to use the proxy device. To build the proxy device add the
```

Comment (pjaaskel) [reply doc/sphinx/source/android.rst:59]: Use the proxy device for ...  [explain the why].

File: doc/sphinx/source/android.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+and set ``-DBUILD_SHARED_LIBS=YES``. This will build a dynamic library of pocl.
+
+*NOTE:* The proxy driver suffers from the same issues the remote driver has with :ref:`Mali GPUs<remote-issues-label>`.
+See that section for a workaround.
+
+
+Setting PoCL Environment Variables
+----------------------------------
+
+The easiest way to set PoCL environment variables is to create a native function that calls stdlib.h's setenv function
```
Comment (pjaaskel) [inline doc/sphinx/source/android.rst:71]:
```suggestion
The easiest way to set PoCL environment variables is to create a native function that calls stdlib.h's setenv function.
```

File: doc/sphinx/source/android.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+Using JOCL
+----------
+
+It is possible to use JOCL on Android. However, by default JOCL does not get built for Android. It also doesn't look for libpocl.
```
Comment (pjaaskel) [inline doc/sphinx/source/android.rst:76]: We shall upstream these changes to JOCL, perhaps point to the JOCL branch which are being upstreamed (PR) directly?

**Review by pjaaskel — CHANGES_REQUESTED** [review]

File: doc/sphinx/source/android.rst (hunk 6 above)
Comment (RABijl) [reply doc/sphinx/source/android.rst:79]: I've made a pr and will change the docs to reflect this if it gets accepted

File: doc/sphinx/source/android.rst (hunk 4 above)
Comment (RABijl) [reply doc/sphinx/source/android.rst]: added a bit on the why at the top of the section

File: doc/sphinx/source/android.rst [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+For newer Android versions (Android 12 and up), you also need to add::
+
+        <uses-native-library
+            android:name="libOpenCL.so"
+            android:required="false" />
+
+to the ``<applications>`` element of the `AndroidManifest.xml`
+
+Once you know that your phone comes with an OpenCL library, it possible to use the proxy device. To build the proxy device add the
```
Comment (pjaaskel) [inline doc/sphinx/source/android.rst:62]:
```suggestion
Once you know that your phone comes with an OpenCL library, it's possible to use the proxy device. To build the proxy device add the
```

**Review by pjaaskel — CHANGES_REQUESTED** [review]

**Pull request merged**